### ISSUE TITLE
clean up clippy warnings

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#895]
Fix the clippy warnings that crept in during WYWAE.

Yesterday or the day before brought new clippy warnings. Normally I fix them w/o creating a GitHub issue, but my development cycle was hectic. So now I'm working on other things but don't want to forget.

**Comment by ctm** [comment]
Ugh. [clippy is segfaulting](https://github.com/rust-lang/rust-clippy/issues/8470). That showed up a few days ago but had seemed to go away, although I also backed off how frequently I was running clippy once it started happening. 

Since there are plenty of people reporting this error, I'll just remove the `high priority` label from this issue and fix the clippy warnings when clippy is running nicely again.

**Comment by ctm** [comment]
So, a workaround is simply to run clippy over and over. There may be a better way, but that's easy enough to allow me to clean up a bunch of recently added clippy warnings and also fix some code I wrote recently that triggered old clippy warnings.

I've relabeled this as `high priority` because I really don't want the tech. debt of not fixing these and it appears that clippy may continue to segfault for a while.
